src/scripts/main.js

```diff
@@ -1,5 +1,60 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  // Getting wall and spider elements
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  // Getting spider size
+  const spiderHeight = spider.clientHeight;
+  const spiderWidth = spider.clientWidth;
+
+  // Getting wall size
+  const wallRect = wall.getBoundingClientRect();
+  const borderWidth = parseFloat(
+    window.getComputedStyle(wall).getPropertyValue('border-width'),
+  );
+  // Getting coordinates of the click relative to the wall
+  // const x = e.clientX - wallRect.left;
+  // const y = e.clientY - wallRect.top;
+
+  const x = e.pageX - wallRect.left;
+  const y = e.pageY - wallRect.top;
+
+  // Checking if the click is inside the wall boundaries considering the spider
+  if (x >= 0 && y >= 0 && x <= wallRect.width && y <= wallRect.height) {
```

Comment:
The condition here checks if the click is within the wall's boundaries. However, it doesn't account for the spider's size. Consider adjusting the condition to ensure the spider doesn't overflow the wall when placed at the edges.